Video to depth maps + anime character segmentation. May be useful to create video masks or silhouette-only videos

kabachuha opened the pull request:
![Screenshot 2023-02-15 at 00-03-00 Stable Diffusion](https://user-images.githubusercontent.com/14872007/218862572-8378eebc-0fe2-4a03-b7f9-23823eb955e9.png)

Closes #277 with all points fulfilled.

The examples will come soon™, the ARB model doesn't want to launch on my GPU for some reason (but should on yours), so the generation is slow as heck

kabachuha committed:
add vid2depth

need fixing some thresh modes

kabachuha committed:
fixup threshold return value

kabachuha committed:
adapt_block_size to int

kabachuha committed:
fixup anime return value

kabachuha committed:
ARB: batch size of 1 for now

kabachuha committed:
don't dither in non-thresholding mode

kabachuha committed:
mixed mode: combine char. silh. + env. depth

kabachuha committed:
don't dither blurred edges

hithereai commented:
@kabachuha , UI fixes in:

https://github.com/deforum-art/deforum-for-automatic1111-webui/commit/b6d37de26907004f39a54b47cd782aa45abec5c1
and:
https://github.com/deforum-art/deforum-for-automatic1111-webui/commit/0165d9195f564d984f863c07782ab902dfbfbc8d

I didn't want to push directly do your PR.

kabachuha commented:
Anime character segmentation-only test

https://user-images.githubusercontent.com/14872007/219113632-781b90a0-fa21-4b08-ab30-836b195c2f53.mp4

kabachuha commented:
## Other video (mixed mode)

### Simple threshold

https://user-images.githubusercontent.com/14872007/219114839-f764baa0-eff3-47b2-9b3f-9f151c17d0f5.mp4

### Simple threshold (auto)

https://user-images.githubusercontent.com/14872007/219114999-e72c8e6d-7b1e-4c9c-a381-44acb0ee1004.mp4

I think we need the upperbound cutoff too as some areas like noise seem to be interpreted as foreground and we ideally need the object in focus to show in the videos

kabachuha committed:
fixed ui

Co-Authored-By: hithereai <121192995+hithereai@users.noreply.github.com>

kabachuha committed:
Update args.py

Co-Authored-By: hithereai <121192995+hithereai@users.noreply.github.com>

kabachuha committed:
customize max thresh value

kabachuha committed:
track mode in out filenames

kabachuha committed:
cosmetic fixup

kabachuha committed:
don't threshold segmentation output

it's already pretty well defined, and the depth cutoff settings definitely shouldn't apply to it

kabachuha committed:
Merge branch 'automatic1111-webui' into bad_apple!!

kabachuha committed:
args for depth norm control

kabachuha committed:
Revert "args for depth norm control"

This reverts commit 5ad939713b24a99e581994d627f835e9a736afb8.

kabachuha committed:
Merge branch 'automatic1111-webui' into bad_apple!!

kabachuha committed:
add 'None' mode to just threshold grayscale

kabachuha committed:
don't load models on None mode

kabachuha committed:
fixup reference in None mode
